src/main/java/org/opensearch/sdk/ExtensionRestPathRegistry.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opensearch.common.path.PathTrie;
+import org.opensearch.rest.RestUtils;
+import org.opensearch.rest.RestRequest.Method;
+
+/**
+ * This class registers REST paths from extension Rest Handlers.
+ */
+public class ExtensionRestPathRegistry {
+
+    // PathTrie to match paths to handlers
+    private PathTrie<ExtensionRestHandler> pathTrie = new PathTrie<>(RestUtils.REST_DECODER);
+    // List to return registered handlers
+    private List<String> registeredPaths = new ArrayList<>();
```

Comment:
This gives off a code smell where this class is additional functionality to the data structure, can we add `getKeys()` to the PathTrie implementation?

Reply:
@peternied I tried that route.  Unfortunately the `PathTrie` does not have functionality to list keys.  You can insert, but the retrieve functionality requires you to give it a path (and possibly parameters to match).  There is no way to generate a list of paths (short of a brute force dictionary attack of all possible strings to see which ones work).

This lack of functionality on the PathTrie is also a challenge in testing (no way to validate/confirm that a path has been inserted, other than an IT) and might be a reasonable feature request to OpenSearch.  But for now, for the purposes of deserializing the extension REST API to pass it over, this slightly smelly code is really the only reasonable option.